docs/DeveloperGuide.md

```diff
@@ -1,33 +1,244 @@
 # Developer Guide
 
-## Design & implementation
+## Table of Contents
+
+[Introduction](#introduction) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;[Purpose](#purpose) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;[Scope](#scope) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;[Design Goals](#design-goals) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;[Maintainability](#maintainability) <br>
+[Definitions](#definitions) <br>
+[Setting up the project in your computer](#setting-up-the-project-in-your-computer) <br>
+[System Overview](#system-overview) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;[Technological view](#technological-view) <br>
+[Architecture](#architecture) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;[Project Overview](#project-overview) <br>
+[Writings class family](#writings-class-family) <br>
+[Filter words class family](#filter-words-class-family) <br>
+[Bunny class family](#bunny-class-family) <br>
+[Names class family](#names-class-family) <br>
+[Design and Implementation](#design-and-implementation) <br>
+[Product scope](#product-scope) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;[Target user profile](#target-user-profile) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;[Value proposition](#value-proposition) <br>
+[User Stories](#user-stories) <br>
+[Non-Functional Requirements](#non-functional-requirements) <br>
+[Glossary](#glossary) <br>
+[Instruction for manual testing](#instructions-for-manual-testing) <br>
```

Comment:
Maybe using bullet points (*) will make the doc more readable?

Reply:
I agree, I like the indentation to signal subheadings but bullet points will increase readability.